Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -1,856 +1,643 @@
+                traverse = false;
+            }
 
-            if (isVisible(visibilityMask)) {
-                flag |= Cesium3DTileChildrenVisibility.VISIBLE;
+            if (traverse) {
+                refines = updateAndPushChildren(tileset, tile, stack, frameState) && parentRefines;
             }
 
-            if (!child.insideViewerRequestVolume(frameState)) {
-                if (isVisible(visibilityMask)) {
-                    flag |= Cesium3DTileChildrenVisibility.VISIBLE_NOT_IN_REQUEST_VOLUME;
-                }
-                visibilityMask = CullingVolume.MASK_OUTSIDE;
-            } else {
-                flag |= Cesium3DTileChildrenVisibility.IN_REQUEST_VOLUME;
-                if (isVisible(visibilityMask)) {
-                    flag |= Cesium3DTileChildrenVisibility.VISIBLE_IN_REQUEST_VOLUME;
+            if (hasEmptyContent(tile)) {
+                // Add empty tile just to show its debug bounding volume
+                addEmptyTile(tileset, tile, frameState);
+                loadTile(tileset, tile, frameState);
+            } else if (add) {
+                // Additive tiles are always loaded and selected
+                selectDesiredTile(tileset, tile, frameState);
+                loadTile(tileset, tile, frameState);
+            } else if (replace) {
+                if (baseTraversal) {
+                    // Always load tiles in the base traversal
+                    // Select tiles that can't refine further
+                    loadTile(tileset, tile, frameState);
+                    if (!refines && parentRefines) {
+                        selectDesiredTile(tileset, tile, frameState);
+                    }
+                } else {
+                    // Load tiles that are not skipped or can't refine further. In practice roughly half the tiles stay unloaded.
+                    // Select tiles that can't refine further. If the tile doesn't have loaded content it will try to select an ancestor with loaded content instead.
+                    if (!refines) { // eslint-disable-line
+                        selectDesiredTile(tileset, tile, frameState);
+                        loadTile(tileset, tile, frameState);
+                    } else if (reachedSkippingThreshold(tileset, tile)) {
+                        loadTile(tileset, tile, frameState);
+                    }
                 }
             }
 
-            child._visibilityPlaneMask = visibilityMask;
+            visitTile(tileset, tile, frameState);
+            touchTile(tileset, tile, frameState);
+            tile._refines = refines;
+            tile._updatedVisibilityFrame = 0; // Reset so visibility is checked during the next pass
         }
+    }
 
-        tile._childrenVisibility = flag;
+    function executeEmptyTraversal(tileset, root, frameState) {
+        // Depth-first traversal that checks if all nearest descendants with content are loaded. Ignores visibility.
```

Comment:
I believe you mentioned you'd like to combine this function with the normal traversal function if possible. Taking a look at both, I would say they are fine being separate as the logic would get too complex.